scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,183 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F", lambda dl, d, du: get_lapack_funcs('gttrf',
```

Comment:
I don't see how this will work.  This lambda function returns the wrapper to gttrf, but I think you mean _execute_ gttrf on the inputs and return the results.

Reply:
Yes, the `(dl, d, du)` is missing. I will add that in.